cpp/src/arrow/io/interfaces.h

```diff
@@ -144,6 +144,15 @@ class ARROW_EXPORT RandomAccessFile : public InputStream, public Seekable {
   /// Necessary because we hold a std::unique_ptr
   ~RandomAccessFile() override;
 
+  /// \brief Create thread-safe InputStream that reads a segment of a
+  /// RandomAccessFile
+  /// \param[in] file a file instance
+  /// \param[in] file_offset the starting position in the file
+  /// \param[in] nbytes the extent of bytes to read
+  /// \param[out] out the created stream
+  static Status GetStream(std::shared_ptr<RandomAccessFile> file, int64_t file_offset,
```

Comment:
Thanks for this patch Wes. Should this be `virtual` instead to make this more flexible? Clients can create custom `RandomAccessFile` and `InputStream` implementations.

Reply:
It's a static function